instrumentation/opentelemetry-instrumentation-httpx/src/opentelemetry/instrumentation/httpx/__init__.py

```diff
@@ -269,7 +270,7 @@ def _extract_parameters(args, kwargs):
         # In httpx >= 0.20.0, handle_request receives a Request object
         request: httpx.Request = args[0]
         method = request.method.encode()
-        url = request.url
+        url = remove_url_credentials(str(request.url))
```

Comment:
Hello - when updating to latest instrumentation, I noticed that `url` stopped being a `httpx.URL` and is a `str` now because of the conversion here. Was it intended? If so, the type is still the old one, though I think `httpx.URL` would be more useful if possible

https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2020/files#diff-eed55d62a8d5f01b055c1bb2d63934c48cfd767416c03826f2c19d4442a76e7bR227

Reply:
No, it wasn't intentional. Moreover, it violates [current type annotation](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/instrumentation/opentelemetry-instrumentation-httpx/src/opentelemetry/instrumentation/httpx/__init__.py#L254). But I don't understand how it may break something if the result of this function always goes through `_prepare_attributes`  function which [converts it to `str`](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/instrumentation/opentelemetry-instrumentation-httpx/src/opentelemetry/instrumentation/httpx/__init__.py#L256) anyway.

Reply:
@ocelotl This is definitely an breaking issue preventing upgrades on my end to v44. We rely on request.url in my request hooks being httpx.URL type

`def request_hook(span: Span, request: Request) -> None` request.url is now a string and it used to be a httpx.URL < v44.

The solution is to wrap it back to httpx.URL:

`httpx.URL(remove_url_credentials(str(request.url)))`